packages/knip/src/WorkspaceWorker.ts

```diff
@@ -341,45 +358,47 @@ export class WorkspaceWorker {
           configFileDir: dirname(configFilePath),
           configFileName: basename(configFilePath),
         };
-        if (hasResolveEntryPaths || hasResolveConfig) {
-          const isManifest = basename(configFilePath) === 'package.json';
-          const fd = isManifest ? undefined : this.cache.getFileDescriptor(configFilePath);
-
-          if (fd?.meta?.data && !fd.changed) {
-            if (fd.meta.data.resolveEntryPaths)
-              for (const id of fd.meta.data.resolveEntryPaths) configEntryPaths.push(id);
-            if (fd.meta.data.resolveConfig) for (const id of fd.meta.data.resolveConfig) addInput(id, configFilePath);
-          } else {
-            const config = await loadConfigForPlugin(configFilePath, plugin, opts, pluginName);
-            const data: CacheItem = {};
-            if (config) {
-              if (hasResolveEntryPaths) {
-                const entryPaths = (await plugin.resolveEntryPaths?.(config, opts)) ?? [];
-                for (const entryPath of entryPaths) configEntryPaths.push(entryPath);
-                data.resolveEntryPaths = entryPaths;
-              }
-              if (hasResolveConfig) {
-                const inputs = (await plugin.resolveConfig?.(config, opts)) ?? [];
-                for (const input of inputs) {
-                  if (isConfig(input)) {
-                    handleConfigInput(input.pluginName, { ...input, containingFilePath: configFilePath });
-                  } else {
-                    addInput(input, configFilePath);
-                  }
-                }
-                data.resolveConfig = inputs;
+
+        let loadedConfig: unknown;
+
+        if (plugin.resolveEntryPaths) {
+          if (!loadedConfig) loadedConfig = await loadConfigForPlugin(configFilePath, plugin, opts, pluginName);
+          if (loadedConfig) {
+            const inputs = await plugin.resolveEntryPaths(loadedConfig, opts);
+            for (const input of inputs) addInput(input, configFilePath);
+          }
+        }
+
+        if (plugin.resolveConfig) {
+          if (!loadedConfig) loadedConfig = await loadConfigForPlugin(configFilePath, plugin, opts, pluginName);
+          if (loadedConfig) {
```

Comment:
Seeing that both `plugin.resolveEntryPaths` and `plugin.resolveConfig` attempt to define `loadedConfig` makes me think they are mutually exclusive properties (don't make sense when specified together in `plugin`). Is that true? If so, maybe it could be helpful to model the type of `plugin` to eliminate the invalid state where both can be passed on the same time. I'm just spitballing here really and could be completely off, also this could not matter at all :) I kind of lack the bigger picture. I'll comment on everything where I have some thought. Feel free to disregard any  of my comments

Reply:
`resolveEntryPaths` and `resolveConfig` can both be implemented as they server different purposes. They might get merged one day now that we have all the "inputs" (`onEntry` etc), but that'd be for another day. So what I was trying to do here is "load if need, and skip if not".